src/app/page/component/page/cyoa-card-collection-page/cyoa-card-collection-page.component.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+  ViewEncapsulation
+} from '@angular/core';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { Observable, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import {
+  CYOAPageCard,
+  CyoaCardCollectionPage
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+import { PageService } from '../../../service/page-service.service';
+
+@Component({
+  selector: 'app-cyoa-card-collection-page',
+  templateUrl: './cyoa-card-collection-page.component.html',
+  styleUrls: ['../default-page.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class CYOACardCollectionComponent implements OnChanges, OnDestroy {
+  @Input() page: CyoaCardCollectionPage;
+  @Input() order: number;
+  @Input() totalPages: number;
+
+  readonly _unsubscribeAll: Subject<void>;
+  private _page: CyoaCardCollectionPage;
+  cards: CYOAPageCard[];
+  ready: boolean;
+  dir$: Observable<string>;
+  formAction$: Observable<string>;
+  isForm$: Observable<boolean>;
+  isModal$: Observable<boolean>;
+  currentYear = new Date().getFullYear();
+  isLastCard: boolean;
+  isFirstCard: boolean;
+  totalCards: number;
+  showBackButton: boolean;
+
+  constructor(
+    readonly pageService: PageService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {
+    this._unsubscribeAll = new Subject<void>();
+    this.dir$ = this.pageService.pageDir$;
+    this.isForm$ = this.pageService.isForm$;
+    this.isModal$ = this.pageService.isModal$;
+    this.formAction$ = this.pageService.formAction$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'page':
+            if (
+              !changes['page'].previousValue ||
+              changes['page'].currentValue !== changes['page'].previousValue
+            ) {
+              this.ready = false;
+              this._page = this.page;
+              this.cards = [];
+              this.init();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    }
+  }
+  currentCardIndex = 0;
```

Comment:
This shouldn't be here, it should either be in a function, like the `init()` or it should be with `showBackButton` at the top